test-utils/testlib/src/node/mod.rs

```diff
@@ -92,7 +92,7 @@ pub trait Node: Send + Sync {
     }
 }
 
-impl Node {
+impl dyn Node {
```

Comment:
`dyn` defines dynamic dispatch. What does `dyn` do in `impl dyn Node`? @bowenwang1996

Reply:
I think nothing. It is required because `Node` is a trait

Reply:
Why is it required? I was not able to find any reference to `impl dyn <trait>`, btw. We can write `impl Node` without `dyn` and would still compile and I pretty sure it will be the same code.

Reply:
Yeah I think so. It's just that there is a compiler warning if I don't add `dyn`.

Reply:
So I talked to the Rust folks. `impl dyn Trait {}` != `impl Trait {}`. And in our case it seems like the only reason why we have methods not defined directly in `pub trait Node: Send + Sync {}` they way `view_balance` is implemented is because these are static methods. And the default implementation of static methods is not allowed inside the trait. So we have to use `impl dyn Trait { <static methods> }`. However if we ever try to call these methods from a normal object and not a trait object we would have to create a duplicate `impl Trait { <same methods>}`.

Reply:
Given how `Node` is used I think we are fine. If we need to use them on normal objects we might as well make the methods non-static.